core/src/test/scala/unit/kafka/server/metadata/BrokerMetadataPublisherTest.scala

```diff
@@ -142,4 +157,80 @@ class BrokerMetadataPublisherTest {
     new TopicsImage(idsMap.asJava, namesMap.asJava)
   }
 
+  class TestBrokerMetadataPublisher(
+    conf: KafkaConfig,
+    metadataCache: KRaftMetadataCache,
+    logManager: LogManager,
+    replicaManager: ReplicaManager,
+    groupCoordinator: GroupCoordinator,
+    txnCoordinator: TransactionCoordinator,
+    clientQuotaMetadataManager: ClientQuotaMetadataManager,
+    featureCache: FinalizedFeatureCache,
+    dynamicConfigHandlers: Map[String, ConfigHandler],
+    private val _authorizer: Option[Authorizer]
+  ) extends BrokerMetadataPublisher(conf,
+      metadataCache,
+      logManager,
+      replicaManager,
+      groupCoordinator,
+      txnCoordinator,
+      clientQuotaMetadataManager,
+      featureCache,
+      dynamicConfigHandlers,
+      _authorizer)
+  {
+    val numTimesReloadCalled = new AtomicInteger(0)
+
+    override def reloadUpdatedFilesWithoutConfigChange(props: Properties): Unit = {
```

Comment:
It feels like we could probably do this with a Mockito spy.

Reply:
done